method/maple/maple.r

```diff
@@ -0,0 +1,205 @@
+
+out_dir <- opt$out_dir
+
+# Output files
+label_file <- file.path(out_dir, "domains.tsv")
+embedding_file <- file.path(out_dir, "embedding.tsv")
+# if additional output files are required write it also to out_dir
+
+# Use these filepaths as input ...
+coord_file <- opt$coordinates
+feature_file <- opt$features
+observation_file <- opt$observations
+
+if (!is.na(opt$neighbors)) {
+  neighbors_file <- opt$neighbors
+}
+if (!is.na(opt$matrix)) {
+  matrix_file <- opt$matrix
+}
+if (!is.na(opt$dim_red)) {
+  dimred_file <- opt$dim_red
+}
+if (!is.na(opt$image)) {
+  image_file <- opt$image
+}
+if (!is.na(opt$config)) {
+  config_file <- opt$config
+}
+
+technology <- opt$technology
+n_clusters <- opt$n_clusters
+
+
+# You can get SpatialExperiment directly
+get_SpatialExperiment <- function(
+    feature_file,
+    observation_file,
+    coord_file,
+    matrix_file = NA,
+    reducedDim_file = NA,
+    assay_name = "counts",
+    reducedDim_name = "PCs") {
+  rowData <- read.delim(feature_file, stringsAsFactors = FALSE, row.names = 1)
+  colData <- read.delim(observation_file, stringsAsFactors = FALSE, row.names = 1)
+
+  coordinates <- read.delim(coord_file, sep = "\t", row.names = 1)
+  coordinates <- as.matrix(coordinates[rownames(colData), ])
+  coordinates[,c(1:2)] <- as.numeric(coordinates[,c(1:2)])
+
+    
+    if (!is.na(reducedDim_file)) {
+    dimRed <- read.delim(reducedDim_file, stringsAsFactors = FALSE, row.names = 1)
+    #reducedDim(spe, reducedDim_name) <- as.matrix(dimRed[colnames(spe), ])
+  }
+  spe <- SpatialExperiment::SpatialExperiment(
+    rowData = rowData, colData = colData, spatialCoords = coordinates,
+      reducedDims = list(pca = as.matrix(dimRed[rownames(colData), ]))
+  )
+
+  if (!is.na(matrix_file)) {
+    assay(spe, "counts", withDimnames = FALSE) <- as(Matrix::t(Matrix::readMM(matrix_file)), "CsparseMatrix")
+    assay(spe, "logcounts", withDimnames = FALSE) <- as(Matrix::t(Matrix::readMM(matrix_file)), "CsparseMatrix")
+  }
+
+  # Filter features and samples
+  if ("selected" %in% colnames(rowData(spe))) {
+    spe <- spe[as.logical(rowData(spe)$selected), ]
+  }
+  if ("selected" %in% colnames(colData(spe))) {
+    spe <- spe[, as.logical(colData(spe)$selected)]
+  }
+
+  
+  return(spe)
+}
+
+# Seed
+seed <- opt$seed
+set.seed(seed)
+# TODO if the method requires the seed elsewhere please pass it on
+# You can use the data as SpatialExperiment
+spe <- get_SpatialExperiment(feature_file = feature_file,observation_file = observation_file,
+                                    coord_file = coord_file, reducedDim_file = dimred_file, matrix_file = matrix_file)
+
+## Your code goes here
+# TODO
+seurat_obj <- as.Seurat(spe)
+
+# Insert image coordinates
+seurat_obj@images$image =  new(
+  Class = 'SlideSeq',
+  assay = "Spatial",
+  key = "coordinates_",
+  coordinates = as.data.frame(spatialCoords(spe))
+)
+
+# Run maple
+maple_results <- fit_maple(
+  seurat_obj, 
+  K = n_clusters, 
+  emb = "PCs", 
```

Comment:
Is this using the provided PCA or recalculate it. Because when add ing the PCA it is called "pca" not "PCs"

Reply:
It does not recalculate the PCA. When emb = "PCs", the function checks if PCs are present, and if not it returns an error message.